packages/entity-server/src/app/createApp.ts

```diff
@@ -33,45 +34,63 @@ import { attachRelationshipsContext } from '../routes/hierarchy/relationships/mi
  * Set up express server with middleware,
  */
 export function createApp() {
-  return new MicroServiceApiBuilder(new TupaiaDatabase())
-    .useBasicBearerAuth('entity-server')
-    .use<SingleEntityRequest | MultiEntityRequest>('hierarchy/:hierarchyName', attachCommonContext)
-    .use<MultiEntityRequest>('hierarchy/:hierarchyName/descendants', attachMultiEntityContext)
-    .use<MultiEntityRequest>('hierarchy/:hierarchyName/relatives', attachMultiEntityContext)
-    .use<MultiEntityRequest>('hierarchy/:hierarchyName/relationships', attachMultiEntityContext)
-    .post<MultiEntityDescendantsRequest>(
-      'hierarchy/:hierarchyName/descendants',
-      handleWith(MultiEntityDescendantsRoute),
-    )
-    .post<MultiEntityRelativesRequest>(
-      'hierarchy/:hierarchyName/relatives',
-      handleWith(MultiEntityRelativesRoute),
-    )
-    .use<MultiEntityRelationshipsRequest>(
-      'hierarchy/:hierarchyName/relationships',
-      attachRelationshipsContext,
-    )
-    .post<MultiEntityRelationshipsRequest>(
-      'hierarchy/:hierarchyName/relationships',
-      handleWith(MultiEntityRelationshipsRoute),
-    )
-    .use<SingleEntityRequest>('hierarchy/:hierarchyName/:entityCode', attachSingleEntityContext)
-    .get<SingleEntityRequest>('hierarchy/:hierarchyName/:entityCode', handleWith(SingleEntityRoute))
-    .get<DescendantsRequest>(
-      'hierarchy/:hierarchyName/:entityCode/descendants',
-      handleWith(EntityDescendantsRoute),
-    )
-    .get<RelativesRequest>(
-      'hierarchy/:hierarchyName/:entityCode/relatives',
-      handleWith(EntityRelativesRoute),
-    )
-    .use<RelationshipsRequest>(
-      'hierarchy/:hierarchyName/:entityCode/relationships',
-      attachRelationshipsContext,
-    )
-    .get<RelationshipsRequest>(
-      'hierarchy/:hierarchyName/:entityCode/relationships',
-      handleWith(EntityRelationshipsRoute),
-    )
-    .build();
+  return (
+    new MicroServiceApiBuilder(new TupaiaDatabase())
+      .useBasicBearerAuth('entity-server')
+
+      // MultiEntity routes
+      .post<MultiEntityRequest>(
+        'hierarchy/:hierarchyName$',
+        attachCommonContext,
+        attachMultiEntityContext,
+        handleWith(MultiEntityRoute),
+      )
```

Comment:
@edmofro Thanks for this suggestion, I really love it ❤️ 

I decided to go with being explicit and defining all middlewares required for each route per route. While this is a bit more verbose it has a few advantages over using any url matching logic:
- It makes it more obvious which route uses which middleware, which as you mentioned was getting quite confusing
- It allows us to ensure the type of the Request matches the middleware for each route. Previously we were using the `SingleEntityRequest` and `MultiEntityRequest` to add the middlewares, where now we use the request of the route itself, which may differ from the base types. This added strictness is really nice imo!